scripts/jobs/customer_services/vonage_landing_to_raw.py

```diff
@@ -1,213 +1,158 @@
-    return subfolders
-
-
-def get_latest_file(list_of_import_years: list) -> str:
-    list_of_raw_dates = []
-
-    if len(list_of_import_years) == 0:
-        return None
-    else:
-        for subfolder in list_of_import_years:
-            path_dictionary = dict(subfolder)
-
-            importstring = path_dictionary["Key"]
-
-            importstring = re.sub(string=importstring,
-                                  pattern="[0-9]{4}-[0-9]{2}-[0-9]{2}\/",
-                                  repl="")
 
-            importstring = re.search("[0-9]{4}-[0-9]{2}-[0-9]{2}", importstring).group()
+    list_of_prefixes = []
+    for dictionary in subfolders:
+        list_of_prefixes.append(dictionary['Prefix'])
 
+    return list_of_prefixes
 
-            list_of_raw_dates.append(importstring)
 
-        list_of_raw_dates = sorted(list_of_raw_dates, key=lambda date: datetime.strptime(date, "%Y-%m-%d"),
-                                   reverse=True)
+def return_largest_prefix(prefix_list):
+    prefix_list.sort(reverse=True)
+    return prefix_list[0]
 
-        largest_value = list_of_raw_dates[0]
-        return largest_value
 
+def get_latest_partition(s3_client, bucket, prefix):
+    # This function looks weird but it drills down to the latest partition without listen every file.
+    # Finds latest Year -> Month -> Day -> Date
+    subfolders = list_subfolders_in_directory(s3_client, bucket, prefix)
+    largest_prefix = return_largest_prefix(subfolders)
 
-def get_latest_value(list_of_import_years: list) -> str:
-    list_of_raw_dates = []
+    subfolders = list_subfolders_in_directory(s3_client, bucket, largest_prefix)
+    largest_prefix = return_largest_prefix(subfolders)
 
-    if len(list_of_import_years) == 0:
-        return None
-    else:
-        for subfolder in list_of_import_years:
-            path_dictionary = dict(subfolder)
+    subfolders = list_subfolders_in_directory(s3_client, bucket, largest_prefix)
+    largest_prefix = return_largest_prefix(subfolders)
 
-            importstring = path_dictionary["Prefix"]
+    subfolders = list_subfolders_in_directory(s3_client, bucket, largest_prefix)
+    largest_prefix = return_largest_prefix(subfolders)
 
-            importstring = re.search("[0-9]*\/$", importstring).group()
+    return largest_prefix
 
-            importstring = re.sub(string=importstring,
-                                  pattern="[^0-9.]".format(),
-                                  repl="")
-            list_of_raw_dates.append(importstring)
 
-        list_of_raw_dates = sorted(list_of_raw_dates, key=int, reverse=True)
-
-        largest_value = list_of_raw_dates[0]
-        return largest_value
-
-
-def get_latest_raw_zone_partition_date(s3_client, bucket, prefix):
-    # Get Year
-    folder_path = f'{prefix}/'
-
-    year_subfolders = list_subfolders_in_directory(s3_client, bucket, folder_path)
+def read_and_write_partition(landing_zone_bucket, partition_path, raw_zone_bucket, raw_zone_prefix):
```

Comment:
This would be hard to test. Maybe read the frame, add the new columns, write the frame rather than wrapping it all in one?

Reply:
I could seperate the functions so that its the one function does one thing rule. I was super wary of memory though. Ill change this